server/jms/src/main/java/org/hawkular/apm/server/jms/TraceStoreMDB.java

```diff
@@ -58,23 +58,24 @@ public void init() {
         setRetryPublisher(tracePublisher);
         setTypeReference(new TypeReference<java.util.List<Trace>>() {
         });
-    }
 
-    /* (non-Javadoc)
-     * @see org.hawkular.apm.server.jms.BulkProcessingMDB#bulkProcess(java.lang.String, java.util.List, int)
-     */
-    @Override
-    protected void bulkProcess(String tenantId, List<Trace> items, int retryCount) throws Exception {
-        long startTime=0;
-        if (perfLog.isLoggable(Level.FINEST)) {
-            startTime = System.currentTimeMillis();
-            perfLog.finest("Performance: about to store trace (first id="+items.get(0).getId()+")");
-        }
-        traceService.storeTraces(tenantId, items);
-        if (perfLog.isLoggable(Level.FINEST)) {
-            perfLog.finest("Performance: store trace (first id="+items.get(0).getId()+") duration="+
-                        (System.currentTimeMillis()-startTime)+"ms");
-        }
+        setProcessor(new AbstractProcessor<Trace, Void>(ProcessorType.ManyToMany) {
+
+            @Override
+            public List<Void> processManyToMany(String tenantId, List<Trace> items) throws Exception {
+                long startTime = 0;
+                if (perfLog.isLoggable(Level.FINEST)) {
+                    startTime = System.currentTimeMillis();
+                    perfLog.finest("Performance: about to store trace (first id=" + items.get(0).getId() + ")");
+                }
+                traceService.storeTraces(tenantId, items);
```

Comment:
If there's an exception here, you'd not have the next log with the duration. Perhaps wrap this on a try-finally? Also, I think this could be better suited for an interceptor with `@AroundInvoke`. This way, you can reuse the logic on other critical places.